### PR TITLE
Refactor Transaction to match spec

### PR DESCRIPTION
**Pull request by MavenRain**
This PR refactors `Transaction` in the `many-types` module to match specification, as cited in https://github.com/liftedinit/many-rs/issues/207

### CODE REVIEW

**Review by fmorency — CHANGES_REQUESTED** [review]
Thanks oni!

I recommend using `None` instead of `Default` for `request` and `response`.

We're still on the fence about `Default` since a P0 issue was caused by it.

@hansl There's a circular dependency between `many-protocol` and `many-types` if `RequestMessage` and `ResponseMessage` stay in `many-protocol`. Do you have a preference about what to move where?

**Comment by hansl** [comment]
If you want to eliminate `many-protocol` itself and move everything in `many-types`, that would make sense.

**Comment by MavenRain** [comment]
> If you want to eliminate `many-protocol` itself and move everything in `many-types`, that would make sense.

Great, I can do that.

File: src/many-types/src/blockchain.rs [hunk 1]
```diff
@@ -83,8 +83,11 @@ pub struct Transaction {
     #[n(0)]
     pub id: TransactionIdentifier,
 
-    #[cbor(n(1), with = "minicbor::bytes")]
-    pub content: Option<Vec<u8>>,
+    #[n(1)]
+    pub request: Option<RequestMessage>,
+
+    #[n(2)]
+    pub response: Option<ResponseMessage>,
```
Comment (MavenRain) [inline src/many-types/src/blockchain.rs:90]: This is the key change of the PR.

File: src/many-types/src/lib.rs [hunk 2]
```diff
@@ -510,6 +519,65 @@ impl<'b, C> Decode<'b, C> for AttributeRelatedIndex {
     }
 }
 
+pub fn decode_request_from_cose_sign1(
```
Comment (MavenRain) [inline src/many-types/src/lib.rs:522]: These were moved from `many-protocol`.

**Review by fmorency — CHANGES_REQUESTED** [review]
Hey,

`WORKSPACE.bazel` still references `many-protocol`, i.e., the Bazel build is broken.

`README.md` still references `many-protocol`

`src/many-types/BUILD.bazel` still references `many-protocol`

File: README.md [hunk 3]
```diff
@@ -39,42 +39,41 @@ graph TD;
...
+  many-error --> many-types;
+  many-identity --> many-types;
   many-error --> many-server;
   many-identity --> many-server;
   many-identity-dsa --> many-server;
   many-macros --> many-server;
   many-modules --> many-server;
-  many-protocol --> many-server;
+  many-types --> many-server;
   many-types --> many-server;
```
[the remark this answers is not in the document]
Comment (MavenRain) [reply README.md:77]: On it

**Comment by MavenRain** [comment]
> Hey,
> 
> `WORKSPACE.bazel` still references `many-protocol`, i.e., the Bazel build is broken.
> 
> `README.md` still references `many-protocol`
> 
> `src/many-types/BUILD.bazel` still references `many-protocol`

Ugh . . . I had these changes made locally, and thought that I pushed, but forgot to save those files.  I'll push them now.

**Review by fmorency — COMMENTED** [review]
Thanks!

Can you add the modified Bazel lock files as well, i.e., 

```
$ CARGO_BAZEL_REPIN=workspace bazel sync --only=crate_index
```

and push `Cargo.Bazel.lock` and `cargo-bazel-lock.json`

**Comment by fmorency** [comment]
Oh, and the Bazel build is still broken

```
➜  many-rs git:(oobi/transaction-impl-to-spec) ✗ bazel build //...:all-targets                            
ERROR: /home/fmorency/dev/many-rs/src/many-identity-dsa/BUILD.bazel:36:13: in rust_library rule //src/many-identity-dsa:many-identity-dsa-for-test: target '//src/many-types:many-types-for-test' is not visible from target '//src/many-identity-dsa:many-identity-dsa-for-test'. Check the visibility declaration of the former target if you think the dependency is legitimate
ERROR: /home/fmorency/dev/many-rs/src/many-identity-dsa/BUILD.bazel:36:13: Analysis of target '//src/many-identity-dsa:many-identity-dsa-for-test' failed
ERROR: Analysis of target '//src/many-identity-hsm:many-identity-hsm-for-test' failed; build aborted: 
INFO: Elapsed time: 5.397s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (434 packages loaded, 10670 targets configured)

```

**Comment by fmorency** [comment]
One last thing, we are using a fork model. Can you fork the repo and work on a branch on your fork, i.e., NOT a branch upstream?

@hansl it appears we can still push branches upstream. Can you disable that, please?

**Comment by MavenRain** [comment]
Closing to move to a branch of the fork, instead of having a pull off of the upstream branch

**Pull request closed without merging**